src/inference/src/ie_layouts.cpp

```diff
@@ -298,19 +298,19 @@ BlockingDesc::BlockingDesc(const SizeVector& blocked_dims,
     this->offsetPaddingToData = dimOffsets;
 
     // check that strides are valid
-    {
-        size_t denseStride = 1;
-
-        for (size_t i = 1; i <= strides.size(); i++) {
-            if (denseStride > strides[strides.size() - i]) {
-                IE_THROW() << "Stride in " << (strides.size() - i)
-                           << "-th dimension "
-                              "is not valid; actual "
-                           << strides[strides.size() - i] << ", should be >= " << denseStride << std::endl;
-            }
-            denseStride = std::max(strides[strides.size() - i], denseStride) * blocked_dims[blocked_dims.size() - i];
-        }
-    }
+    // {
+    //     size_t denseStride = 1;
+
+    //     for (size_t i = 1; i <= strides.size(); i++) {
+    //         if (denseStride > strides[strides.size() - i]) {
+    //             IE_THROW() << "Stride in " << (strides.size() - i)
+    //                        << "-th dimension "
+    //                           "is not valid; actual "
+    //                        << strides[strides.size() - i] << ", should be >= " << denseStride << std::endl;
+    //         }
+    //         denseStride = std::max(strides[strides.size() - i], denseStride) * blocked_dims[blocked_dims.size() - i];
+    //     }
+    // }
```

Comment:
why do we need it ?

Reply:
@ilya-lavrenov @ilyachur 
Here I am trying to fix the problem of **tensor_to_blob** (which we discussed in email thread) by copying it to intel_cpu folder and fix it.
https://github.com/openvinotoolkit/openvino/pull/18476/files#diff-408940c3c9d85d477330292dc8adb22f70ff541e5ca28576bafc7b780f6612bfR123-R243

A typical usage is to wrap an custom ITensor implementation (ref: https://github.com/openvinotoolkit/openvino/pull/18476/files#diff-e7cb2f658f935019e3646b615c6968601416946bdf65931e9ae2df12811b26e5R13-R43) whose underlying memory instance is not initialized when it is wrapped, thus strides is 0 and cannot pass the check.

So what would you advice to handle this kind of custom ITensor wrap to blob?

Reply:
> Here I am trying to fix the problem of tensor_to_blob (which we discussed in email thread) by copying it to intel_cpu folder and fix it.

I suppose it's already fixed by https://github.com/openvinotoolkit/openvino/pull/18364, is not it?

Reply:
sounds good. May I have it exported as an external function?

And what's your advice to this change?
> A typical usage is to wrap an custom ITensor implementation (ref: https://github.com/openvinotoolkit/openvino/pull/18476/files#diff-e7cb2f658f935019e3646b615c6968601416946bdf65931e9ae2df12811b26e5R13-R43) whose underlying memory instance is not initialized when it is wrapped, thus strides is 0 and cannot pass the check.

Reply:
Discussed separately:
- we can add a condition here to enable this check only for case of static shapes.
- tensor_to_blob can be moved to plugin API